packages/react-components/src/components/HorizontalGallery.tsx

```diff
@@ -58,15 +62,20 @@ export const HorizontalGallery = (props: HorizontalGalleryProps): JSX.Element =>
   const numberOfChildren = React.Children.count(children);
   const lastPage = Math.ceil(numberOfChildren / childrenPerPage) - 1;
 
+  const indexesArray: number[][] = useMemo(() => {
+    return bucketize([...Array(numberOfChildren).keys()], childrenPerPage);
+  }, [numberOfChildren, childrenPerPage]);
+
+  useEffect(() => {
+    if (setTilesToRender && indexesArray) {
+      setTilesToRender(indexesArray[page]);
+    }
+  }, [indexesArray, setTilesToRender, page]);
+
   const paginatedChildren: React.ReactNode[][] = useMemo(() => {
     return bucketize(React.Children.toArray(children), childrenPerPage);
   }, [children, childrenPerPage]);
 
-  // If children per page is 0 or less return empty element
-  if (childrenPerPage <= 0) {
-    return <></>;
-  }
-
   const firstIndexOfCurrentPage = page * childrenPerPage;
   const clippedPage = firstIndexOfCurrentPage < numberOfChildren - 1 ? page : lastPage;
   const childrenOnCurrentPage = paginatedChildren[clippedPage];
```

Comment:
This can be:
```
const childrenOnCurrentPage = useMemo(() => indexesArrays[clippedPage].map(index => {
  return children[index];
}), [ **dependency array here** ])
```

So even we change some lines of code in the future, those children nodes will be generated strictly followed what is in the array

Reply:
And don't forget to remove lines from 75-78